src/org/mozilla/javascript/NativeArray.java

```diff
@@ -2252,6 +2261,18 @@ private static boolean js_isArray(Object o) {
         return "Array".equals(((Scriptable) o).getClassName());
     }
 
+    private static Scriptable js_toReversed(
+            Context cx, Scriptable scope, Scriptable thisObj) {
+        Scriptable o = ScriptRuntime.toObject(cx, scope, thisObj);
+        long len = (getLengthProperty(cx, o) > Integer.MAX_VALUE) ? 0 : getLengthProperty(cx, o);
```

Comment:
i guess getLengthProperty(cx, o) is expensive, maybe calling this only once is a good idea